File: trunk/src/proof-of-concept/Python (wxWindows)/DemoApp/DemoWindow.py
import wx
from XMLBasedGUI.XmlBasedWindow import XmlBasedWindow



class DemoWindow(XmlBasedWindow):
    def __init__(self):
        XmlBasedWindow.__init__(self, "demo.xml")
        
        # bind button event
        button = self.getControlByID("button")
        self.Bind(wx.EVT_BUTTON, self.OnButtonClick, button)

    def OnButtonClick(self, event):
        editbox = self.getControlByID("text")
        wx.MessageDialog(self, editbox.GetValue()).ShowModal()        #, style=wx.MessageDialog.wxOK

File: trunk/src/proof-of-concept/.NET (Windows Forms)/DemoApp/DemoWindow.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Windows.Forms;

namespace DemoApp
{
    class DemoWindow : XMLBasedGUI.XmlBasedWindow
    {
        public DemoWindow() : base("demo.xml")
        {
            Button button = (Button)getControlByID("button");
            if (button != null)
            {
                EventHandler eHandler = new EventHandler(userClicked);
                button.Click += eHandler;
            }
        }

        public void userClicked(object sender, EventArgs e)
        {
            TextBox editbox = (TextBox)getControlByID("text");
            if (editbox != null)
            {
                MessageBox.Show(editbox.Text);
            }
            else
            {
                MessageBox.Show("ERROR!");
            }
        }

    }
}
